distribution/lib/Standard/Base/0.0.0-dev/src/Random.enso

```
@@ -1,99 +1,307 @@
+
+       Arguments:
+       - start_time: The lower bound of the range to pick from (inclusive).
+       - end_time: The upper bound of the range to pick from (inclusive).
+
+       The generated time is at a granularity of 1 second.
+
+       Note: the time range end is inclusive because it is impossible to specify
+       an end time after the last second of the day.
+
+       > Example
+         Generate a random time.
+
+             import Standard.Base.Random.Random
+
+             start = Time_Of_Day.new 8 12 15
+             end = Time_Of_Day.new 9 40 2
+             t = Random.time start end
+    time : Time_Of_Day -> Time_Of_Day -> Time_Of_Day
+    time (start_time:Time_Of_Day=(Time_Of_Day.new 0 0 0)) (end_time:Time_Of_Day=(Time_Of_Day.new 23 59 59)) =
+        Random.default_instance.time start_time end_time
+
+    ## GROUP Random
+       Return a new UUID.
+
+       ! `uuid` does not use the underlying `java.util.Random` instance, and is
+       therefore not affected by the seed.
+
+       > Example
+         Generate a uuid.
+
+             import Standard.Base.Random.Random
+
+             uuid = Random.uuid
+    uuid : Text
+    uuid = Random.default_instance.uuid
+
+    ## GROUP Random
+       Returns a new `Vector` containing a random sample of the input vector,
+       with or without replacement.
+
+       Arguments:
+       - vector: The `Vector` to take items from.
+       - count: The number of items to return.
+       - with_replacement: If false, each value in the `Vector` can only be
+         selected once during a single call; the returned values will contain no
+         duplicates. If true, values from the `Vector` can be returned multiple
+         times and duplicates are possible.
+
+       If `with_replacement` is false, and `count` is equal to or greater than
+       the size of the collection, a random permutation of the collection is
+       returned.
```

I think that this makes sense if the `count` is _equal_ to the size of the collection.

If the `count` is greater, IMHO we should throw an error.

In general, as a user I'd expect `Random.items col n . length == n` to hold. So IMO if we cannot find `n` elements, we should throw. That's just my opinion though and I appreciate there may be arguments for the current behaviour too.

---

I think I agree, but this is following the original design, so I'll leave it as-is and let @jdunkerley tell us his opinion.

---

Actually, I change my mind -- I put in this check. We can take it out later if we want.